### DISCUSSION TITLE
with member cache disabled, why do i have more Member objects in memory than guilds?

### DISCUSSION THREAD

**Discussion by ioistired** [Rapptz/discord.py#5865] [Q&A] [answered]
![image showing that there are 4,177 Member objects in memory, according to objgraph, but only 3166 guilds](https://user-images.githubusercontent.com/538336/94587442-3846d880-0248-11eb-86cb-d40f0e2b0865.png)

Is this a bug? I would expect that with `member_cache_flags=MemberCacheFlags.none()` and `bot.intents.members = False`, that there would be approximately as many Member objects as guilds. Why are there more? I don't use fetch_member or query_members(cache=True) in my code.

**Comment by Rapptz** [comment, marked as answer]
Chances are it's probably the members that are cached from the messages. If you don't care for the message cache you can disable it under `max_messages=None` inside `Client.__init__`.

**Reply by ioistired** [reply]
Thank you. Can the message cache size be changed at runtime or do I have to restart?

**Reply by Rapptz** [reply]
Restarting is the only way.
